### ISSUE TITLE
CBOR NumberConverstion=Double Issue

### ISSUE THREAD

**Issue by cmhofmeister** [peteroupc/CBOR#49]
I am working on an API between a mobile app and BLE device that uses the Zephyr OS.  The implementation does not have support for half float (0xf9).  
To attempt to get around this issue, I set the number conversation to be double.  My understanding is that all numbers should then be forced into a CBOR double (oxfb).
https://peteroupc.github.io/CBOR/docs/PeterO.Cbor.JSONOptions.ConversionMode.html 

```
string msg = {"p1": 220.7351, "p2": 1.0}
JSONOptions jo = new JSONOptions("numberconversion=double");
obj1 = CBORObject.FromJSONString(msg, jo);
byte[] msgBytes = obj1.EncodeToBytes();
//msgBytes =  A2 62 70 31 FB 40 6B 97 85 F0 6F 69 44 62 70 32 F9 3C 00
```

Putting the output into a CBOR viewer (http://cbor.me/) we get the following result:
A2                     # map(2)
   62                  # text(2)
      7031             # "p1"
   FB 406B9785F06F6944 # primitive(4641970442455705924)
   62                  # text(2)
      7032             # "p2"
   **F9** 3C00             # primitive(15360)

What should be noted above is that the 1.0 value is still encoded as a half-float (oxf9), even though the conversation specified it to be a double.

**Comment by peteroupc** [comment]
Putting `numberconversion=double` affects only how `FromJSONString` decodes the JSON into CBOR.  It doesn't affect how that CBOR is later encoded.  For example, you're calling `EncodeToBytes` with no `CBOREncodeOptions`, which means that the default options are used which don't include half-float encoding.

Of the options currently available in `CBOREncodeOptions`, the [`Ctap2Canonical` option](https://peteroupc.github.io/CBOR/docs/PeterO.Cbor.CBOREncodeOptions.html#Ctap2Canonical) writes out CBOR floating-point numbers in their 64-bit encoding form (`double`) regardless of whether they can be encoded in a smaller form.  However, `Ctap2Canonical` also imposes other rules on encoding, which may or may not be desirable, especially if the FIDO Client-to-Authenticator Protocol 2 or Web Authentication (WebAuthn) is not involved.

At the moment, there is no _dedicated_ option for encoding CBOR objects as CBOR without using the half-float encoding.

**Comment by cmhofmeister** [comment]
Thank you for the very clear response and possible options.  Maybe something to consider adding in the future.

**Comment by peteroupc** [comment]
New option Float64 was added to CBOREncodeOptions and expected to be available in version 4.4.
